### DISCUSSION TITLE
【V5】V4 中的 annotation 迁移到 V5

### DISCUSSION THREAD

**Discussion by BinghuiXie** [antvis/G2#5255] [Q&A] [answered]
由于 V5 中把 `annotation` 都收敛到了 `mark` 里面

G2_v4 | G2_v5
-- | --
chart.annotation().arc(option) | chart.arc()
chart.annotation().image(option) | chart.image()
chart.annotation().line(option) | chart.lineX()chart.lineY()
chart.annotation().text(option) | chart.text()
chart.annotation().region(option) | chart.region()chart.regionX()chart.regionY()
chart.annotation().regionFilter(option) | 不直接支持
chart.annotation().dataMarker(option) | 不直接支持
chart.annotation().dataRegion(option) | 不直接支持
chart.annotation().shape(option) | chart.shape()
chart.annotation().html(option) | 不支持

目前看了一下对于 `regionFilter`，`dataMarker` 以及 `dataRegion` 没有直接的 `mark` 可以绘制实现，而 `html` 没有找到能实现的方法，这四种目前是怎么处理的呢？尤其是针对 `html` 类型的

**Comment by pearmini** [comment, marked as answer]
除了 `chart.annotation().regionFilter(option)` 不直接支持，其余的可以如下使用：

- [dataMarker](https://g2.antv.antgroup.com/examples/annotation/text/#line-text)
- dataRegion: `chart.region().label({ text: '' })`
- html: `chart.shape().style({ render: () => {} })` 参考[这个测试](https://github.com/antvis/G2/blob/v5/__tests__/plots/static/income-point-circle.ts)。

`chart.annotation().regionFilter(option)` 的实现可以参考[颜色渐变](https://g2.antv.antgroup.com/examples/general/line/#line-variable-color)。
